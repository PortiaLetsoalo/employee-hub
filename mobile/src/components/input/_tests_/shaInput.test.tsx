import React from 'react';
import { create, act } from 'react-test-renderer';
import ShaInput from '../shaInput';

jest.mock('@rneui/themed', () => ({
  Input: 'Input',
}));

jest.mock('../components/customLabel', () => 'CustomLabel');

const createShaInput = (props) => create(<ShaInput {...props} />);

describe('ShaInput Component', () => {
  let tree;

  beforeEach(() => {
    const defaultProps = {
      label: 'Username',
      required: true,
      onChangeText: jest.fn(),
      value: 'testuser',
      inputStyle: { borderWidth: 1, borderColor: 'gray', borderRadius: 5, padding: 10 },
      containerStyle: { marginBottom: 10 },
      inputContainerStyle: { backgroundColor: 'lightgray', borderRadius: 5, padding: 5 },
      labelStyle: { fontWeight: 'bold', fontSize: 16 },
    };

    tree = createShaInput(defaultProps);
  });

  it('matches snapshot', () => {
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('is defined', () => {
    expect(
      <ShaInput
        label="Username"
        required={true}
        onChangeText={jest.fn()}
        value="testuser"
        inputStyle={{ borderWidth: 1, borderColor: 'gray', borderRadius: 5, padding: 10 }}
        containerStyle={{ marginBottom: 10 }}
        inputContainerStyle={{ backgroundColor: 'lightgray', borderRadius: 5, padding: 5 }}
      />
    ).toBeDefined();
  });

  it('renders with correct props', () => {
    const instance = tree.root;
    expect(instance.props.value).toEqual('testuser');
    expect(instance.props.label).toEqual('Username');
    expect(instance.props.required).toEqual(true);
    expect(instance.props.inputStyle).toEqual({ borderWidth: 1, borderColor: 'gray', borderRadius: 5, padding: 10 });
    expect(instance.props.containerStyle).toEqual({ marginBottom: 10 });
    expect(instance.props.inputContainerStyle).toEqual({ backgroundColor: 'lightgray', borderRadius: 5, padding: 5 });
  });

  it('updates props correctly', () => {
    act(() => {
      tree.update(
        <ShaInput
          label="Email"
          required={false}
          onChangeText={jest.fn()}
          value="test@example.com"
          inputStyle={{ borderWidth: 2, borderColor: 'black', borderRadius: 8, padding: 12 }}
          containerStyle={{ marginTop: 20 }}
          inputContainerStyle={{ backgroundColor: 'white', borderRadius: 8, padding: 8 }}
        />
      );
    });

    const instance = tree.root;
    expect(instance.props.value).toEqual('test@example.com');
    expect(instance.props.label).toEqual('Email');
    expect(instance.props.required).toEqual(false);
    expect(instance.props.inputStyle).toEqual({ borderWidth: 2, borderColor: 'black', borderRadius: 8, padding: 12 });
    expect(instance.props.containerStyle).toEqual({ marginTop: 20 });
    expect(instance.props.inputContainerStyle).toEqual({ backgroundColor: 'white', borderRadius: 8, padding: 8 });
  });
});
