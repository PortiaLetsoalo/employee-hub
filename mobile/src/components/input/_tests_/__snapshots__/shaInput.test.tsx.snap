// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShaInput Component matches snapshot 1`] = `
<Input
  containerStyle={
    Object {
      "marginBottom": 10,
      "paddingHorizontal": 0,
      "width": undefined,
    }
  }
  inputContainerStyle={
    Object {
      "backgroundColor": "lightgray",
      "borderBottomWidth": 0,
      "borderRadius": 5,
      "padding": 5,
    }
  }
  inputStyle={
    Object {
      "backgroundColor": "white",
      "borderColor": "gray",
      "borderRadius": 5,
      "borderWidth": 1,
      "color": "#2C2738",
      "flex": 1,
      "fontFamily": undefined,
      "fontSize": 14,
      "height": 40,
      "margin": 0,
      "padding": 10,
      "paddingHorizontal": 10,
    }
  }
  label={
    <CustomLabel
      label="Username"
      labelStyle={
        Object {
          "color": undefined,
          "display": "flex",
          "fontFamily": undefined,
          "fontSize": 16,
          "fontWeight": "bold",
          "marginBottom": 5,
        }
      }
      required={true}
    />
  }
  labelStyle={
    Object {
      "color": undefined,
      "display": "flex",
      "fontFamily": undefined,
      "fontSize": 16,
      "fontWeight": "bold",
      "marginBottom": 5,
    }
  }
  onChangeText={[MockFunction]}
  placeholderTextColor="#B2B8BE"
  required={true}
  underlineColorAndroid="transparent"
  value="testuser"
/>
`;
